wrangler-core/src/main/java/io/cdap/directives/column/SetHeader.java

```diff
@@ -95,11 +95,17 @@ public Mutation lineage() {
 
   @Override
   public Schema getOutputSchema(SchemaResolutionContext context) {
-    List<Schema.Field> fields = new ArrayList<>(context.getInputSchema().getFields());
+    List<Schema.Field> inputFields = context.getInputSchema().getFields();
+    List<Schema.Field> outputFields = new ArrayList<>();
     for (int i = 0; i < columns.size(); i++) {
-      Schema.Field existing = fields.remove(i);
-      fields.add(i, Schema.Field.of(columns.get(i), existing.getSchema()));
+      if (i < inputFields.size()) {
```

Comment:
nit: this if condition can be added in for loop condition. `for (int i = 0; i < columns.size() && i < inputFields.size(); i++)`